### ISSUE TITLE
reverse DNS names now mandatory / problematic for existing blank entries

### ISSUE THREAD

**Issue by listerr** [inex/IXP-Manager#440]
##### ISSUE TYPE

Bug Report

##### OS
```
4.8.0
```
##### SUMMARY
In the new 4.8.0 interface (vlan interface) the reverse DNS fields are now mandatory, so a record cannot be saved without DNS entries.

Unfortunately we have many records (especially IPv6) which previously had no reverse DNS entry. Therefore we can't save anything to a vlan interface without making up something for the reverse DNS entry if it's blank.

Also the arpa reverse DNS API template generates entries for "."

```
140    IN      PTR     .
142    IN      PTR     .
144    IN      PTR     .
145    IN      PTR     .
```
Can the template skip any blank entries as before, or apply some default such as:

"no-dns-yet-5-57-81.lonap.net"

**Comment by barryo** [comment]
> Can the template skip any blank entries as before, or apply some default such as: 

This commit no longer pulls VLAN interface records with a blank hostname from the database. As you cannot have an ARPA entry without a hostname, it's less a template and more a core functionality issue.

**Comment by barryo** [comment]
> In the new 4.8.0 interface (vlan interface) the reverse DNS fields are now mandatory, so a record cannot be saved without DNS entries.

So this is a case of INEX standard operating procedures clashes with another IX's different procedures.

We don't allow VLAN interfaces without a hostname. Even if the member doesn't ask for it (and they often don't as many as confused by the question). If it's Example Networks Ltd with domain example.com, we'd stick in inex.example.com (or inex-lan1 / inex-lan1-ipv4 / etc).

It's often pained me that there's no forward domain for this but then I guess that's also probably the case for 99% of those that specify a domain.

On balance, I think being able to attribute an IP address to a peering member with a ptr lookup is better than allowing blank names. I think @listerr in the case where you've allowed blanks heretofore, the options are:

1. bulk database update
2. fix on a case by case basis as you edit vlan interfaces with (e.g.) no-dns-yet-5-57-81.lonap.net

**Comment by listerr** [comment]
Seems this is addressed as of 5.3.x IXP Manager, _VLAN Interface: ARPA hostname can be made optional by setting the configuration option_: `IXP_FE_VLANINTERFACES_HOSTNAME_REQUIRED=false`

**Comment by barryo** [comment]
Yes 😄 

Mentioned in the release notes: https://github.com/inex/IXP-Manager/releases/tag/v5.3.0

And docs updated for this also: https://docs.ixpmanager.org/usage/interfaces/#ipv4ipv6-details

I normally hunt open/closed issues also but somehow I didn't dig this one up.
